papis/commands/tag.py

```diff
@@ -0,0 +1,148 @@
+"""
+This command allows users to add and remove tags to documents in their library.
+
+Examples
+^^^^^^^^
+
+- To add a tag use the `--new` (or `-n`) option:
+
+    .. code:: sh
+
+        papis tag --new TAG1 --new TAG2 QUERY
+
+  This will add TAG1 and TAG2 to a document matched by QUERY. You can repeat
+  `--new` as many times as you need. The query is any query supported by papis. 
+  If the query matches more than one document, Papis' picker will be started to 
+  let you pick the document from those matched (just as with Papis' other 
+  commands).
+
+  Tags are only added if they do not already exist, which is to say that the
+  same tag cannot exist more than once in a given document.
+
+- To remove a tag use the `--remove` (or `-r`) option:
+
+    .. code:: sh
+
+        papis tag --remove TAG1 QUERY
+
+  This removes TAG1 from the document. If it didn't exist before, nothing 
+  happens. You can use `--remove` as many times as you like to remove multiple
+  tags.
+
+- Use `--clear` (or `-c`) to remove all tags:
+
+    .. code:: sh
+
+        papis tag --clear --new TAG1 QUERY
+
+  This removes all tags from the document and then adds TAG1. You can of course
+  also use `--clear` by itself to simply remove all tags without adding new 
+  ones.
+
+- Use `--all` (or `-a`) with any of the other options to apply the tagging
+  operation to all matching documents:
+
+    .. code:: sh
+
+        papis tag --all --new TAG1 QUERY
+  
+  This adds TAG1 to all documents matching the QUERY rather than opening the
+  picker to let you choose one.
+
+"""
+from typing import Tuple, Set, List
+
+import click
+
+import papis.config
+import papis.commands
+import papis.document
+import papis.strings
+import papis.cli
+
+logger = papis.logging.get_logger(__name__)
+
+TAG_FORMAT = papis.config.get("tag-format")
+
+
+def make_set_of_tags(tags: List[str] | str) -> Set[str]:
+    if isinstance(tags, list) and TAG_FORMAT == "list":
+        set_of_tags = set(tags)
+    elif isinstance(tags, str) and TAG_FORMAT == "comma_separated":
```

Comment:
This is something I've been thinking of and decided against because the best would then be to allow arbitrary separators. For instance, we might have both '; ' and ';' (so with and without trailing white space). I didn't find any easy way to write trailing white space in the config file (though admittedly I didn't try much), and so opted to just go with descriptive names instead. Do you think it's ok to simply assume (and document) that papis always assumes that the separator is followed by a white space?

Reply:
Hm, we can always support some form of quoting in the value of `tag-format-separator` and do
```
separator = papis.config.getstring("tag-format-separator").strip('"').strip("'")
```
to remove any extra quotes around the string? `ConfigParser` itself seems to indeed to `value.strip()` to remove any whitespace :(

Reply:
Ok, that should work! It's not the prettiest solution, but for sure the most versatile. And then maybe mid term, we can just move everyone to lists